api/HouseInput.cs

```diff
@@ -1,23 +1,106 @@
-﻿/**
+/**
  * Reading device state from the house app.
  */
 using System;
+using System.Threading;
+using System.IO;
+using System.Diagnostics;
+using Newtonsoft.Json;
+using System.Net;
+using System.Threading.Tasks;
+using System.Text;
 
 namespace api
 {
-/**
- * Class which reads state from the house app.
- */
-public class HouseInput : IDeviceInput
-{
-	public HouseInput()
-	{
-	}
+    /**
+     * Class which reads state from the house app.
+     */
+    public class HouseInput : IDeviceInput
+    {
+        public HouseInput(string URL)
+        {
+            _URL = URL;
+        }
 
-	public bool read(Device dev)
+        public bool read(Device dev)
 	{
-		return true;
+		var _ = readHelper(dev);
+
+        return true;
 	}
-}
+
+        private async Task<string> readHelper(Device dev)
+        {
+            try
+            {
+                var request = (HttpWebRequest)WebRequest.Create(_URL);
+                request.Method = "GET";
+                request.ContentType = "application/json";
+
+                try
+                {
+                    using (var stream = await Task<Stream>.Factory.FromAsync(request.BeginGetRequestStream, request.EndGetRequestStream, request))
+                    {
+                        result = new byte[stream.Length];
+                        await stream.ReadAsync(result, 0, (int)stream.Length);
+                    }
+
+                    Interfaces.UpdateDevice(dev, result.ToString());
+                }
+                catch (Exception ex)
+                {
+                    _StreamException = ex;
+                    return null;
+                }
+
+                try
+                {
+
+                    WebResponse responseObject = await Task<WebResponse>.Factory.FromAsync(request.BeginGetResponse, request.EndGetResponse, request);
+                    var responseString = responseObject.GetResponseStream();
+                    var streamread = new StreamReader(responseString);
+                    string confirm = await streamread.ReadToEndAsync();
+                    return confirm;
```

Comment:
The first part I'm trying to update a device after getting the input. Then, the other try block, it appears as unnecessary in this case, I'll edit it out.